projects/telemetry/test/test_nmea.c

```diff
@@ -0,0 +1,42 @@
+#include <stddef.h>
+#include <string.h>
+#include "nmea.h"
+#include "unity.h"
+
+void setup_test(void) {}
+void teardown_test(void) {}
+
+void test_compute_checksum(void) {
+  char *message;
+  size_t message_len;
+  char computed_checksum[3];
+
+  message = "$PSRF103,00,01,00,01*";
+  message_len = strlen(message);
+  nmea_compute_checksum(message, message_len, computed_checksum);
+  TEST_ASSERT_EQUAL_STRING("25", computed_checksum);
+
+  message = "$PSRF103,05,00,01,01*";
+  message_len = strlen(message);
+  nmea_compute_checksum(message, message_len, computed_checksum);
+  TEST_ASSERT_EQUAL_STRING("20", computed_checksum);
+
+  message = "$GPGGA,053740.000,2503.6319,N,12136.0099,E,1,08,1.1,63.8,M,15.2,M,,0000*64";
+  message_len = strlen(message);
+  nmea_compute_checksum(message, message_len, computed_checksum);
+  TEST_ASSERT_EQUAL_STRING("64", computed_checksum);
+
+  message = "$GPGLL,2503.6319,N,12136.0099,E,053740.000,A,A*52";
+  message_len = strlen(message);
+  nmea_compute_checksum(message, message_len, computed_checksum);
+  TEST_ASSERT_EQUAL_STRING("52", computed_checksum);
+}
+
+void test_compare_checksum(void) {
+  char *input =
+      "$GPGGA,053740.000,2503.6319,N,12136.0099,E,1,08,1.1,63.8,"
+      "M,15.2,M,,0000*64";
+
+  size_t message_len = strlen(input);
+  TEST_ASSERT_TRUE(nmea_compare_checksum(input, message_len));
```

Comment:
You should try testing some edge cases too - empty string, malformed string, incorrect checksum, etc.